wayshot/src/ext_wayshot.rs

```diff
@@ -0,0 +1,267 @@
+        }
+    }
+}
+
+pub fn ext_capture_toplevel(
+    state: &mut WayshotConnection,
+    use_stdout: bool,
+    pointer: bool,
+) -> Result<(Option<DynamicImage>, String), WayshotImageWriteError> {
+    let toplevels = state.toplevels();
+    let names: Vec<String> = toplevels.iter().map(|info| info.id_and_title()).collect();
+
+    let selection = FuzzySelect::with_theme(&ColorfulTheme::default())
+        .with_prompt("Choose Application")
+        .default(0)
+        .items(&names)
+        .interact()?;
+
+    let toplevel = toplevels[selection].clone();
+    let img = state
+        .ext_capture_toplevel2(pointer.to_capture_option(), toplevel)
+        .map_err(WayshotImageWriteError::WaylandError)?;
+
+    if use_stdout {
+        let stdout = io::stdout();
+        let mut writer = BufWriter::new(stdout.lock());
+        let mut buffer = Cursor::new(Vec::new());
+        img.write_to(&mut buffer, image::ImageFormat::Png)?;
+        writer.write_all(buffer.get_ref())?;
+        writer.flush()?;
+        Ok((None, names[selection].clone()))
+    } else {
+        Ok((Some(img), names[selection].clone()))
+    }
+}
+
+pub fn ext_capture_output(
+    state: &mut WayshotConnection,
+    output: Option<String>,
+    use_stdout: bool,
+    pointer: bool,
+) -> eyre::Result<(Option<image::DynamicImage>, String), WayshotImageWriteError> {
+    let outputs = state.vector_of_outputs();
+    let names: Vec<&str> = outputs.iter().map(|info| info.name.as_str()).collect();
+    let selection = match output {
+        Some(name) => names
+            .iter()
+            .position(|tname| *tname == name)
+            .ok_or(WayshotImageWriteError::OutputNotExist)?,
+        None => FuzzySelect::with_theme(&ColorfulTheme::default())
+            .with_prompt("Choose Screen")
+            .default(0)
+            .items(&names)
+            .interact()?,
+    };
+
+    let output_name = names[selection].to_string();
+    let output = outputs[selection].clone();
+    let img = state
+        .ext_capture_single_output(pointer.to_capture_option(), output)
+        .map_err(WayshotImageWriteError::WaylandError)?;
+
+    if use_stdout {
+        let stdout = io::stdout();
+        let mut writer = BufWriter::new(stdout.lock());
+        let mut buffer = Cursor::new(Vec::new());
+        img.write_to(&mut buffer, image::ImageFormat::Png)?;
+        writer.write_all(buffer.get_ref())?;
+        writer.flush()?;
+        Ok((None, output_name))
+    } else {
+        Ok((Some(img), output_name))
+    }
+}
+
+pub fn ext_capture_area(
+    state: &mut WayshotConnection,
+    use_stdout: bool,
+    pointer: bool,
+) -> Result<(Option<DynamicImage>, WayshotResult), WayshotImageWriteError> {
+    let (data, img_width, img_height, _color_type, region) =
+        state.ext_capture_area2(pointer.to_capture_option(), |w_conn: &WayshotConnection| {
+            let info = libwaysip::get_area(
+                Some(libwaysip::WaysipConnection {
```

Comment:
libwaysip has changed something these days, but it should be quite easy to change